Find a functionally equivalent code:android.hardware.SystemSensorManager.TriggerEventQueue.dispatchSensorEvent:COMMENT
Method Modifier: protected   
Comment:// Called from native code.

@@ -1,6 +1,9 @@
 {
     final Sensor sensor = sHandleToSensor.get(handle);
-    TriggerEvent t = mTriggerEvents.get(handle);
+    TriggerEvent t = null;
+    synchronized (mTriggerEvents) {
+        t = mTriggerEvents.get(handle);
+    }
     if (t == null) {
         Log.e(TAG, "Error: Trigger Event is null for Sensor: " + sensor);
         return;

